### PR TITLE
sanity checks (fixes #37)

### LINKED ISSUE #37: Out of bound panic
* [X ] I've read and understood the [Contributing guidelines](https://github.com/slackhq/go-audit/blob/master/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [ X] I've read and agree to the [Code of Conduct](https://github.com/slackhq/go-audit/blob/master/CODE_OF_CONDUCT.md).
* [ X] I've searched for any related issues and avoided creating a duplicate issue.

#### Description

Had once a panic in logs that came from an out of bound array in parser.go.

#### Reproducible in:

`go-audit` version: HEAD
OS version(s): Linux Ubuntu 16.04

#### Steps to reproduce:
??

### PR DESCRIPTION
**Pull request by stephane-martin**
* [X ] I've read and understood the [Contributing guidelines](https://github.com/slackhq/go-audit/blob/master/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [X ] I've read and agree to the [Code of Conduct](https://github.com/slackhq/go-audit/blob/master/CODE_OF_CONDUCT.md).
* [X ] I've been mindful about doing atomic commits, adding documentation to my changes, not refactoring too much.
* [X ] I've a descriptive title and added any useful information for the reviewer. Where appropriate, I've attached a screenshot and/or screencast (gif preferrably).
* [X ] I've written tests to cover the new code and functionality included in this PR (no need)
* [X ] I've read, agree to, and signed the [Contributor License Agreement (CLA)](https://docs.google.com/a/slack-corp.com/forms/d/1q_w8rlJG_x_xJOoSUMNl7R35rkpA7N6pUkKhfHHMD9c/viewform).

#### PR Summary
Some sanity checks for parser.go

#### Related Issues
fixes #37

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -64,6 +64,7 @@ func NewAuditMessage(nlm *syscall.NetlinkMessage) *AuditMessage {
 
 // Gets the timestamp and audit sequence id from a netlink message
 func parseAuditHeader(msg *syscall.NetlinkMessage) (time string, seq int) {
+	var err error
```
Comment (glaslos) [inline parser.go]: How about returning the error as well? If you return an error, you don't have to reset `time` and `seq` as the consumer can check the error value.

File: parser.go [hunk 2]
```diff
@@ -74,14 +75,24 @@ func parseAuditHeader(msg *syscall.NetlinkMessage) (time string, seq int) {
...
-
-		// Remove the header from data
-		msg.Data = msg.Data[headerStop+3:]
+		seq, err = strconv.Atoi(header[sep+1:])
+		if err != nil {
+			time = ""
+			seq = 0
+			return
+		}
+		if len(msg.Data) < (headerStop + 3) {
```
Comment (glaslos) [inline parser.go]: There should be an explanation for `+3`

**Comment by stephane-martin** [comment]
@glaslos agreed, but I tried to keep the PR minimal, compared to original code (which I don't understand fully). I hope the main authors will check it.

**Comment by nbrownus** [comment]
Thanks for the PR! I'll be looking this over later today. I would also like to hear more about your setup and, if possible, see the log line or raw data that caused the panic.

**Pull request closed without merging**